src/components/file-browser/file-browser-hooks.ts

```diff
@@ -0,0 +1,81 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { FileBrowserBackend, IncrementalResponse } from 'src/components/file-browser/file-browser-backends'
+import { useCallbackOne } from 'use-memo-one'
+
+
+type GetIncrementalResponse<T> = (options: { signal: AbortSignal }) => Promise<IncrementalResponse<T>>
+
+export const useIncrementalResponse = <T>(getFirstPage: GetIncrementalResponse<T>) => {
```

Comment:
my first thought was to recommend moving this and the GetIncrementalResponse type to a new directory src/components/utils since it is a general reusable hook... but then I asked the question, why does this need to be a hook?

...and my impression is that the state changes like isLoading and results are assumed to trigger a re-render of the visual component that is using this hook.  I think we can separate these concerns.  I will connect offline to discuss.

Reply:
Added the `LoadedState` type and refactored this hook to return a `state` value of type `LoadingState`.